### ISSUE TITLE
OctopOS main program

### ISSUE THREAD

**Issue by JoshuaHassler** [SPACE-HAUC/OctopOS#5]
The main program needs to be written for octopOS. It should:
1. Setup an instance of octopOS
2. Fork and exec each module
    * Each module needs to have its MSGKEY passed as argv[0]
3. Create a tentacle and a thread for each child that listens for messages
4. Register a handler for SIGCHLD, this will be expanded on with issue #4

**Comment by JoshuaHassler** [comment]
It could be worthwhile to write the main function as a sperate project and have it link octopOS as a library. what do you think @LLazarek?

**Comment by LLazarek** [comment]
Yes, I think that's a good idea. Ideally what we're using octopOS for (as the satellite driver) is a separate concern from octopOS itself. We may need to do some refactoring to enable that kind of usage, but I would say it's worth a shot.

**Comment by JoshuaHassler** [comment]
cmake already compiles octopOS as a library, so it shouldn't be bad. I'll look into configuring cmake's `make install` to automate library installation. I think we should hijack the CDH Main repo for this.
